### ISSUE TITLE
CachedDbAcl must check if the $entity has the method bindNode()

### ISSUE THREAD

**Issue by Xety** [cakephp/acl#50]
Hello,

I'm trying to use the `CachedDbAcl` Adapter and there are an issue.

I'm using the configuration with `GroupOnly` as mentioned in the doc.
But the Adapter create cache file for `$alias . '_' . $ref[$name]['id']` as you can see here :
![vdx1auo](https://cloud.githubusercontent.com/assets/8210023/6068734/eb8d76ee-ad7e-11e4-9779-7ec2f41e21cc.png)

https://github.com/cakephp/acl/blob/master/src/Adapter/CachedDbAcl.php#L101

Instead of that, it must check if a `bindNode()` exist in the entity, like in the node() method :
https://github.com/cakephp/acl/blob/master/src/Model/Table/AclNodesTable.php#L114-L139

I have added this code to the method `_getNodeCacheKey()` :
```php
protected function _getNodeCacheKey($ref)
{
	if (empty($ref)) {
		return '';
	} elseif (is_string($ref)) {
		return Inflector::slug($ref, '_');
	} elseif (is_object($ref) && $ref instanceof Entity) {
		return $ref->source() . '_' . $ref->id;
	} elseif (is_array($ref) && !(isset($ref['model']) && isset($ref['foreign_key']))) {
		$name = key($ref);
		list(, $alias) = pluginSplit($name);

		$bindTable = TableRegistry::get($name);
		$entityClass = $bindTable->entityClass();

		if ($entityClass) {
			$entity = new $entityClass();
		}

		if (empty($entity)) {
			throw new Exception\Exception(__d('cake_dev', "Entity class {0} not found in CachedDbAcl::_getNodeCacheKey() when trying to bind {1} object", [$type, $this->alias()]));
		}

		$tmpRef = null;
		if (method_exists($entity, 'bindNode')) {
			$tmpRef = $entity->bindNode($ref);
		}

		if (empty($tmpRef)) {
			$ref = [
				'model' => $alias,
				'foreign_key' => $ref[$name][$bindTable->primaryKey()]
			];
		} else {
			$ref = $tmpRef;
		}

		return $ref['model'] . '_' . $ref['foreign_key'];
	} elseif (is_array($ref)) {
		return $ref['model'] . '_' . $ref['foreign_key'];
	}

	return '';
}
```

And now it work as expected (Group only):
![pf0utlr](https://cloud.githubusercontent.com/assets/8210023/6068871/1975b32c-ad80-11e4-9d6e-9ced372dd522.png)

**Comment by dakota** [comment]
Would you be able to make a pull request with this change?

**Comment by Xety** [comment]
Hmm no because i have some issues with tests.

**Comment by Xety** [comment]
Closed as #70 is open now.
